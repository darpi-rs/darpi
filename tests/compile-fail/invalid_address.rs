use darpi::{app, handler};

#[handler]
pub(crate) async fn home() -> String {
    format!("home")
}

#[darpi::main]
async fn main() -> Result<(), darpi::Error> {
    env_logger::builder().is_test(true).try_init().unwrap();

    app!({
        address: "127.0.0.1:foo",
        handlers: [{
            route: "/",
            method: GET,
            handler: home
        }]
    })
    .run()
    .await
}
