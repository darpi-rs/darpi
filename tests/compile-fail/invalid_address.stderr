error: invalid server address: `127.0.0.1:foo` error: `invalid IP address syntax`
  --> tests/compile-fail/invalid_address.rs:13:18
   |
13 |         address: "127.0.0.1:foo",
   |                  ^^^^^^^^^^^^^^^
